### PR TITLE
feat: support update over binary field

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

**Comment by eddyxu** [comment]
Pending lance release with https://github.com/lancedb/lance/pull/2579

File: python/python/lancedb/util.py [hunk 1]
```diff
@@ -231,6 +222,15 @@ def _(value: str):
     return f"'{value}'"
 
 
+@value_to_sql.register(bytes)
+def _(value: bytes):
```
Comment (chebbyChefNEQ) [inline python/python/lancedb/util.py:226]: do we want to support more types? like str?

**Review by chebbyChefNEQ — APPROVED** [review]

File: python/python/lancedb/util.py (hunk 1 above)
Comment (wjones127) [reply python/python/lancedb/util.py:226]: ? `str` is the function above.

Comment (eddyxu) [reply python/python/lancedb/util.py:226]: string is already supports (see 2 lines above)

**Review by wjones127 — APPROVED** [review]

File: python/python/lancedb/util.py (hunk 1 above)
Comment (chebbyChefNEQ) [reply python/python/lancedb/util.py:226]: I meant coercing str to bytes automatically. nvm doesn't seem possible from this interface

**Pull request merged**